src/pylibsshext/scp.pyx

```diff
@@ -131,14 +133,17 @@ cdef class SCP:
             if rc == libssh.SSH_ERROR:
                 raise LibsshSCPException("Failed to start read request: %s" % self._get_ssh_error_str())
 
-            # Read the file
-            rc = libssh.ssh_scp_read(scp, read_buffer, size)
-            if rc == libssh.SSH_ERROR:
-                raise LibsshSCPException("Error receiving file data: %s" % self._get_ssh_error_str())
-
-            py_file_bytes = read_buffer[:size]
             with open(local_file, "wb") as f:
-                f.write(py_file_bytes)
+                remaining_bytes_to_read = size
+                while remaining_bytes_to_read > 0:
+                    # Read the file
```

Comment:
Comments repeating what the code does are pointless. Proper identifier names should be used instead.
```suggestion
```
Learning materials: https://www.youtube.com/watch?v=YklKUuDpX5c / https://www.youtube.com/watch?v=x-kB2o8sd5c / https://www.youtube.com/watch?v=1__lNTlj1_w.

Reply:
Removed.